.github/workflows/formatting.yml

```diff
@@ -26,3 +26,24 @@ jobs:
       - name: Check formatting
         working-directory: ${{runner.workspace}}/nrn
         run: external/coding-conventions/bin/format -v --dry-run
+
+      # If formatting fails, apply formatting and push changes.
+      # This will trigger another workflow run, which will cancel the current one.
+      - name: Apply formatting
+        working-directory: ${{runner.workspace}}/nrn
+        if: failure() && github.event_name == 'pull_request'
+        run: |
+          # Checkout PR
+          gh pr checkout ${{ github.event.pull_request.number }}
+
+          # Apply formatting
+          external/coding-conventions/bin/format -v
+
+          # Commit & push changes
+          git config --global user.name "github-actions[bot]"
+          git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          git add -u :/
+          git commit -a -m "Fix formatting"
+          git push
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Note: A personal access token is required for gh cli, GITHUB_TOKEN doesn't work by design

Reply:
@alkino maybe you can pick this up

Reply:
we need to update GitHub token with someone else?

Reply:
IIRC events triggered by the GITHUB_TOKEN,  except workflow_dispatch and repository_dispatch, will not create a new workflow run. So autoformat commit -> no CI update (azure works fine I believe). Therefore a PAT is needed